src/dwarf/Gfde.c

```diff
@@ -138,6 +139,15 @@ parse_cie (unw_addr_space_t as, unw_accessors_t *a, unw_word_t addr,
         augstr[i++] = ch;
     }
 
+  if (version > 3)
```

Comment:
Are we parsing the version correctly? I am asking because, I tried adding `printf(">>> the version is: %d\n", version);` before this condition and did:

```sh
./autogen.sh && make && make check

readelf --debug-dump=info tests/.libs/Ltest-mem-validate | grep -A 2 'Compilation Unit @'
```
it shows:
```
  Compilation Unit @ offset 0x0:
   Length:        0xe36 (32-bit)
   Version:       4
```

but when executing it shows, `>>> the version is: 1`

Same result if when explicitly setting dwarf version to 5:
```sh
autoreconf -i && ./configure CFLAGS="-gdwarf-5" CXXFLAGS="-gdwarf-5" && make -j && make check
```
readelf shows `Version:       5` but our dwarf parser still gets version 1.

and same thing happens if `CC=$(command clang) CXX=$(command clang++)`

does that mean the dwarf version parsing is wrong?

Reply:
Yea I noticed this as well, I always see 1.  I'll dig in to it when I get a chance

Reply:
If this version is supposed to be the same as Compilation Unit version from the readelf output above, then the first difference I noticed is that we are reading one byte for version, while binutils uses two bytes (`unsigned short`).

Definition: https://github.com/bminor/binutils-gdb/blob/3451a2d7a3501e9c3fc344cbc4950c495f30c16d/binutils/dwarf.h#L59
Reading: https://github.com/bminor/binutils-gdb/blob/3451a2d7a3501e9c3fc344cbc4950c495f30c16d/binutils/dwarf.c#L3561